tree/ntuple/v7/src/RField.cxx

```diff
@@ -1078,14 +1092,25 @@ void ROOT::Experimental::RVectorField::ReadGlobalImpl(NTupleSize_t globalIndex,
    RClusterIndex collectionStart;
    fPrincipalColumn->GetCollectionInfo(globalIndex, &collectionStart, &nItems);
 
-   auto oldNItems = typedValue->size() / fItemSize;
-   for (std::size_t i = nItems; i < oldNItems; ++i) {
-      auto itemValue = fSubFields[0]->CaptureValue(typedValue->data() + (i * fItemSize));
-      fSubFields[0]->DestroyValue(itemValue, true /* dtorOnly */);
-   }
-   typedValue->resize(nItems * fItemSize);
-   for (std::size_t i = oldNItems; i < nItems; ++i) {
-      fSubFields[0]->GenerateValue(typedValue->data() + (i * fItemSize));
+   if (fSubFields[0]->GetTraits() & kTraitTrivialType) {
+      typedValue->resize(nItems * fItemSize);
+   } else {
+      const auto oldNItems = typedValue->size() / fItemSize;
```

Comment:
We should add explicitly the description of 'when' we construct/destruct.  (for example besides treating the trivial case, this also introduce more constructor/destruction that before but it is not easy to see that from the code).

Reply:
(And actually I am confused on whether or not the implementation is what we meant to do, i.e. it "seems" to do too many destruction or not enough)

Reply:
I added the explanation in the new architecture.md file. Note that we currently don't copy/move objects when the collection buffer is reallocated. That's probably a todo.